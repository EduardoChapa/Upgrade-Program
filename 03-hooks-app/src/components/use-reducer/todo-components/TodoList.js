import React from 'react';
import TodoListItem from './TodoListItem';

function TodoList({ handleToggle, handleDelete, todos}) {
    return (
        <div className='col-7'>
            <ul className='list-group list-group-flush'>
                { todos.map((todo, i) => <TodoListItem key={ i } todo={ todo } index={ i } handleToggle={ handleToggle } handleDelete={ handleDelete } /> ) }
            </ul>
        </div>
    );
}

export default TodoList;