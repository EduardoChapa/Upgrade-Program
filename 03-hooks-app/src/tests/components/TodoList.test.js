import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '../../components/use-reducer/todo-components/TodoList';
import todos from '../fixtures/todos';

describe('Testing TodoList component', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(<TodoList todos={ todos } handleDelete={ handleDelete } handleToggle={ handleToggle }/>);

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have two TodoItemList components', () => {
        expect(wrapper.find('TodoListItem').length).toBe(todos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    });
});