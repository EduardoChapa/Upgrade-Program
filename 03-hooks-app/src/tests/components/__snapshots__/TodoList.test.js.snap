// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoList component should display correctly 1`] = `
<div
  className="col-7"
>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="0"
      todo={
        Object {
          "desc": "Testing",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="1"
      todo={
        Object {
          "desc": "Testing again",
          "done": true,
          "id": 2,
        }
      }
    />
  </ul>
</div>
`;
